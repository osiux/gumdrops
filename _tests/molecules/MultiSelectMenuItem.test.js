/* globals mount */
import React from 'react';
import MultiSelectMenuItem from '../../components/molecules/MultiSelectMenuItem';

const subOptions = [
    { name: 'Lassie', value: 1, selected: false },
    { name: 'Snoopy', value: 2, selected: false },
    { name: 'Toto', value: 3, selected: false },
    { name: 'Brian Griffin', value: 3, selected: false }
];

const defaultProps = {
    callback: () => {},
    index: 1,
    isActive: false,
    name: 'Foo',
    onChange: () => {},
    onClick: () => {},
    selected: false,
    size: 'xs',
    value: 0
};

describe('Expect <MultiSelectMenuItem>', () => {
    it('to render', () => {
        const wrapper = mount(<MultiSelectMenuItem {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to call the onChange prop handler when options change', () => {
        const props = {
            ...defaultProps,
            onChange: jest.fn(),
            callback: jest.fn()
        };
        const wrapper = mount(<MultiSelectMenuItem {...props} />);
        wrapper
            .find('Checkbox input[type="checkbox"]')
            .first()
            .simulate('change');

        expect(props.onChange).toBeCalledWith({
            index: props.index,
            selected: !props.selected,
            value: props.value
        });

        expect(props.callback).toBeCalledWith(props.index, props.value, !props.selected); // deprecated
    });

    it('to call the onChange prop handler with sub options', () => {
        const props = {
            ...defaultProps,
            subOptions,
            onChange: jest.fn(),
            callback: jest.fn()
        };
        const wrapper = mount(<MultiSelectMenuItem {...props} />);
        wrapper
            .find('Checkbox input[type="checkbox"]')
            .first()
            .simulate('change');

        expect(props.onChange).toBeCalledWith({
            index: props.index,
            selected: !props.selected,
            value: props.value,
            options: subOptions
        });

        expect(props.callback).toBeCalledWith(props.index, props.value, !props.selected); // deprecated
    });
});
