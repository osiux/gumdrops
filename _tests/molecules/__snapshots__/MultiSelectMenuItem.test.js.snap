// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <MultiSelectMenuItem> to render 1`] = `
<MultiSelectMenuItem
  callback={[Function]}
  index={1}
  isActive={false}
  name="Foo"
  onChange={[Function]}
  onClick={[Function]}
  selected={false}
  size="xs"
  value={0}
>
  <li
    className="gds-multi-select__menu-item"
    onClick={[Function]}
  >
    <div
      className="gds-multi-select__menu-link"
    >
      <div
        className="gds-form-group gds-multi-select__option"
      >
        <Checkbox
          checked={false}
          label="Foo"
          onChange={[Function]}
          size="xs"
        >
          <div
            className="gds-form-group__checkbox gds-form-group__checkbox--xs"
          >
            <label
              className="gds-form-group__checkbox-label"
            >
              <input
                checked={false}
                className="gds-form-group__checkbox-input"
                onChange={[Function]}
                type="checkbox"
              />
              <span
                className="gds-form-group__checkbox-indicator"
              />
              Foo
            </label>
          </div>
        </Checkbox>
      </div>
    </div>
  </li>
</MultiSelectMenuItem>
`;
